src/gluonnlp/data/qa_preprocessing_utils.py

```diff
@@ -0,0 +1,144 @@
+"""Utility classes and functions for qa data processing"""
+
+import collections
+
+
+def truncate_seq_pair(tokens_a, tokens_b, max_length):
+    """Truncates a sequence pair in place to the maximum length."""
+    # This is a simple heuristic which will always truncate the longer sequence
+    # one token at a time. This makes more sense than truncating an equal percent
+    # of tokens from each, since if one sequence is very short then each token
+    # that's truncated likely contains more information than a longer sequence.
+    while True:
+        total_length = len(tokens_a) + len(tokens_b)
+        if total_length <= max_length:
+            break
+        if len(tokens_a) > len(tokens_b):
+            tokens_a.pop()
+        else:
+            tokens_b.pop()
+
+
+def truncate_seqs_equal(seqs, max_length):
```

Comment:
Would the code be easier to understand with a small recursion? I'm considering a function like the following

``` python
import numpy.ma as ma


def truncate(lens, maxlen):
    def recursive(lens, maxlen):
        argmin = lens.argmin()
        minval = lens[argmin]
        quotient, remainder = divmod(maxlen, len(lens) - sum(lens.mask))
        if minval <= quotient:  # Ignore values that don't need truncation
            lens.mask[argmin] = 1
            lens = recursive(lens, maxlen - minval)
            lens.mask[argmin] = 0
        else:  # Truncate all
            lens.data[~lens.mask] = [
                quotient + 1 if i < remainder else quotient
                for i in range(lens.count())
            ]

        return lens

    if sum(lens) <= maxlen:
        return lens

    lens = ma.masked_array(lens, mask=[0] * len(lens))
    return recursive(lens, maxlen).data.tolist()
```